src/compile/repeat.ts

```diff
@@ -0,0 +1,295 @@
+import {extend, keys, vals, flatten, duplicate, mergeDeep, contains, forEach, Dict} from '../util';
+import {VgData, VgLegend} from '../vega.schema';
+import {ORDINAL} from '../type';
+import {LAYOUT} from '../data';
+
+import {parseAxisComponent} from './axis';
+import {buildModel} from './common';
+import {assembleData, parseRepeatData} from './data/data';
+import {assembleLayout, parseRepeatLayout} from './layout';
+import {Model} from './model';
+import {parseScaleComponent, ScaleComponents} from './scale';
+
+export type RepeatValues = {
+  row: string,
+  column: string
+}
+
+export class RepeatModel extends Model {
+  private _repeat: Repeat;
+
+  private _children: Model[];
+
+  constructor(spec: RepeatSpec, parent: Model, parentGivenName: string, repeatValues: RepeatValues) {
+    super(spec, parent, parentGivenName, repeatValues);
+
+    // Config must be initialized before child as it gets cascaded to the child
+    const config = this._config = this._initConfig(spec.config, parent);
+
+    const repeat  = this._repeat = spec.repeat;
+    this._children = this._initChildren(spec, repeat, repeatValues);
+    this._scale  = this._initScale(repeat, config);
+    this._axis = {};
+  }
+
+  private _initConfig(specConfig: Config, parent: Model) {
+    return mergeDeep(duplicate(defaultConfig), specConfig, parent ? parent.config() : {});
+  }
+
+  private _initScale(repeat: Repeat, config: Config): Dict<Scale> {
+    return [ROW, COLUMN].reduce(function(_scale, channel) {
+      if (repeat[channel]) {
+
+        _scale[channel] = extend({
+          type: ScaleType.ORDINAL,
+          round: config.facet.scale.round,
+          domain: repeat[channel],
+
+          padding: 60
+        }, {});
+      }
+      return _scale;
+    }, {} as Dict<Scale>);
+  }
+
+  private _initChildren(spec: RepeatSpec, repeat: Repeat, repeatValues: RepeatValues): Model[] {
+    let children = [];
+    const row = repeat.row || [repeatValues ? repeatValues.row : null];
+    const column = repeat.column || [repeatValues ? repeatValues.column : null];
+
+    // cross product
+    for (let r = 0; r < row.length; r++) {
+      const rowField = row[r];
+      for (let c = 0; c < column.length; c++) {
+        const columnField = column[c];
+
+        const name = (rowField ? '_' + rowField : '') + (columnField ? '_' + columnField : '');
+
+        const childRepRepeatValues = {
+          row: rowField,
+          column: columnField
+        };
+        children.push(buildModel(spec.spec, this, this.name('child' + name), childRepRepeatValues));
+      }
+    }
+
+    return children;
+  }
+
+  public repeat() {
+    return this._repeat;
+  }
+
+  public hasMultipleDimensions() {
+    return this.has(ROW) && this.has(COLUMN);
+  }
+
+  public has(channel: Channel): boolean {
+    return !!this._repeat[channel];
+  }
+
+  public children() {
+    return this._children;
+  }
+
+  public dataTable(): string {
+    return this.has(ROW) ? 'fields_row' : 'fields_column';
```

Comment:
This returns the data table for the repeat. It's wrong now but it was right when we still had the cross. Good catch.